src/celeritas/optical/detail/OpticalGenAlgorithms.hh

```diff
@@ -44,6 +44,23 @@ remove_if_invalid(Collection<celeritas::optical::GeneratorDistributionData,
                   size_type,
                   StreamId);
 
+//---------------------------------------------------------------------------//
+// Count the number of optical photons in the distributions.
+size_type
+count_num_photons(Collection<celeritas::optical::GeneratorDistributionData,
+                             Ownership::reference,
+                             MemSpace::host> const&,
```

Comment:
Perhaps it would be helpful to define
```c++
template<MemSpace M>
using GeneratorDistributionRef = Collection<::celeritas::optical::GeneratorDistributionData, Ownership::reference, M>;
```
in this header file

Reply:
Yeah that looks a bit neater.